client/mysqlbinlog.cc

```diff
@@ -744,9 +748,12 @@ static void convert_path_to_forward_slashes(char *fname)
 */
 static bool shall_skip_database(const char *log_dbname)
 {
-  return one_database &&
-         (log_dbname != NULL) &&
-         strcmp(log_dbname, database);
+  if(log_dbname == NULL)
+    return false;
+
+  if (one_database)
```

Comment:
It seems to me given your current implementation that we don't need this variable anymore. We can just replace its usage with `std::set`'s `.empty()` function. Let me know if I'm mistaken.